src/tools/illink/src/ILLink.Shared/DataFlow/ValueSet.cs

```diff
@@ -15,6 +15,8 @@ namespace ILLink.Shared.DataFlow
 	public readonly struct ValueSet<TValue> : IEquatable<ValueSet<TValue>>, IEnumerable<TValue>, IDeepCopyValue<ValueSet<TValue>>
 		where TValue : notnull
 	{
+		const int MaxValuesInSet = 1024;
```

Comment:
I would make it even lower - tracking values is pretty expensive (all of the Meet/DeepCopy stuff we do for arrays). I guess my main point is "If the code is going to cause the analyzer to see 100s of values, how much does that differ from 1000s" - I don't think the difference is interesting as in I don't think there's a case in between which we really care about supporting.
And if we're going to give up, we should do so fast - based on my debugging the original issue, we spend a LOT of time in the `left.DeepCopy` and `right.DeepCopy` and `values.UnionWith` right before the limit here. In the repro case both of the values are big (1000s of items) and the union is expensive.

So I would vote to lower the limit to something like 256 or maybe 512.